golem-worker-executor-base/src/services/scheduler.rs

```diff
@@ -172,22 +202,47 @@ impl SchedulerServiceDefault {
             })
             .collect::<Vec<_>>();
 
-        let mut owned_worker_ids = HashSet::new();
+        // ! Do not exist early from this loop because of failed actions, as it will cause all other actions to be skipped.
+        // ! Errors will only be logged anyway, so just log them inline here and ignore.
         for (key, action) in matching {
-            owned_worker_ids.insert(action.owned_worker_id().clone());
-            self.key_value_storage
-                .with_entity("scheduler", "process", "scheduled_action")
-                .remove_from_sorted_set(KeyValueStorageNamespace::Schedule, key, &action)
-                .await?;
-
-            match action {
-                ScheduledAction::CompletePromise { promise_id, .. } => {
-                    self.promise_service
-                        .complete(promise_id, vec![])
-                        .await
-                        .map_err(|golem_err| format!("{golem_err}"))?;
+            match action.clone() {
+                ScheduledAction::CompletePromise {
+                    promise_id,
+                    account_id,
+                } => {
+                    let owned_worker_id = OwnedWorkerId::new(&account_id, &promise_id.worker_id);
+
+                    let result = self
+                        .promise_service
+                        .complete(promise_id.clone(), vec![])
+                        .await;
+
+                    // TODO: We probably need more error handling here as not completing a promise that is expected to complete can lead to deadlocks.
+                    match result {
+                        Ok(_) => {
+                            // activate worker so it starts processing the newly completed promises
```

Comment:
I didn't understand this part either. How is this related to scheduled invocation? (Couldn't spot in PR review)

Reply:
Unrelated to scheduled invocation, just general notes regarding the current schedule implementation